src/PrestaShopBundle/Form/Admin/Sell/Product/ShippingType.php

```diff
@@ -0,0 +1,196 @@
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Sell\Product;
+
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\PositiveOrZero;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+use PrestaShopBundle\Form\Admin\Type\TranslatableType;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\MoneyType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\Type;
+
+/**
+ * Form type containing product shipping information
+ */
+class ShippingType extends TranslatorAwareType
+{
+    /**
+     * @var string
+     */
+    private $currencyIsoCode;
+
+    /**
+     * @var FormChoiceProviderInterface
+     */
+    private $carrierChoiceProvider;
+
+    /**
+     * @var FormChoiceProviderInterface
+     */
+    private $deliveryTimeNoteTypesProvider;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param array $locales
+     * @param string $currencyIsoCode
+     * @param FormChoiceProviderInterface $carrierChoiceProvider
+     * @param FormChoiceProviderInterface $additionalDeliveryTimeNoteTypesProvider
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        string $currencyIsoCode,
+        FormChoiceProviderInterface $carrierChoiceProvider,
+        FormChoiceProviderInterface $additionalDeliveryTimeNoteTypesProvider
+    ) {
+        parent::__construct($translator, $locales);
+        $this->currencyIsoCode = $currencyIsoCode;
+        $this->carrierChoiceProvider = $carrierChoiceProvider;
+        $this->deliveryTimeNoteTypesProvider = $additionalDeliveryTimeNoteTypesProvider;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('width', TextType::class, [
```

Comment:
Why `TextType` instead of `NumberType` ?

Reply:
Number type converts it to float, but we use string everywhere, so I thought it would avoid some casting and maybe float calculation problems :man_shrugging: 
But now I think its better to use Number and cast to float, because with text user might want to use comma to separate decimals and it will break :D